src/Data/Path/Pathy.purs

```diff
@@ -478,6 +476,18 @@ module Data.Path.Pathy
   parseAbsDir :: String -> Maybe (AbsDir Unsandboxed)
   parseAbsDir = parsePath (const Nothing) (const Nothing) (const Nothing) Just
 
+  -- | Folds a `Path` into a value using two specified functions and a provided
+  -- | default value.
+  -- |
+  -- | The first function folds a `FileName` into our value, and the second
+  -- | function folds a `DirName` into our value.
+  fold :: forall c a b s. (FileName -> c -> c) -> (DirName -> c -> c) -> c -> Path a b s -> c
+  fold f g z (Current      ) = z
+  fold f g z (Root         ) = z
+  fold f g z (ParentIn p   ) = fold f g z p
```

Comment:
@jdegoes So do you think it's cool to merge for now, and revisit what you had in mind later?